plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorService.java

```diff
@@ -77,9 +81,11 @@ public Void call() {
 
     @Override
     protected Runnable wrap(final Runnable r) {
+        Metrics.metricRegistry().counter(QUEUE_SIZE_METRIC).inc();
         return () -> {
+            Metrics.metricRegistry().counter(QUEUE_SIZE_METRIC).dec();
             ThreadContext context = setUp();
-            try {
+            try (var timer = Metrics.metricRegistry().timer(EXECUTION_DURATION_METRIC).time()) {
```

Comment:
Here, I think the `Timer` is fine. We mainly want to see if tasks are on average taking a very long time, for example because of poor IO performance or something like that. I am also not sure though if the global metric offers significant benefit over the per-build timings, so perhaps this should be dropped.